src/maingoblint.ml

```diff
@@ -155,18 +156,13 @@ let basic_preprocess ~all_cppflags fname =
   (* The actual filename of the preprocessed sourcefile *)
   let nname =  Filename.concat !Goblintutil.tempDirName (Filename.basename fname) in
   if Sys.file_exists (get_string "tempDir") then
-    nname
+    (nname, None)
   else
     (* Preprocess using cpp. *)
     (* ?? what is __BLOCKS__? is it ok to just undef? this? http://en.wikipedia.org/wiki/Blocks_(C_language_extension) *)
```

Comment:
Not really part of this PR, but we should take the opportunity to update this comment. It is indeed the thing that is linked and seems to be unavailable in GCC (except for GCC variants shipped by Apple). 

https://clang.llvm.org/docs/BlockLanguageSpec.html

Reply:
I've always wondered what exactly is this undefine achieving, what problem is it solving and whether it's still necessary. Because we only undefine it for basic preprocessing, but not compilation database or cilly.

Reply:
Someone who uses OS X would have to investigate that. @vogler @vesalvojdani ?

Reply:
It was introduced in [fd29d6d6873e5833d58c64cd1a32a02271142de6](https://github.com/goblint/analyzer/commit/fd29d6d6873e5833d58c64cd1a32a02271142de6#diff-c4a7423227c515488a65063285c91d2defd87ab7962f4ed66c2f76bd4f728f98R149) without any explanation when the standards for Goblint weren't as high.

Reply:
There's also this weird page for some CIL-based tool that suggests a bunch of OSX workarounds: http://tigress.cs.arizona.edu/machine.html.

Reply:
Moved to #605.